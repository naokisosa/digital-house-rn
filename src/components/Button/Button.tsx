import { StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';
import React from 'react';
import BaseText from '@components/BaseText/BaseText';
import { colors } from '@style';

interface Props {
  title: string;
  onPress: () => void;
  style?: ViewStyle;
}

const Button = ({ onPress, title, style }: Props) => {
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      <BaseText style={styles.title}>{title}</BaseText>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primary,
    borderRadius: 10,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontWeight: '800',
    color: colors.white,
    fontSize: 16,
    lineHeight: 21.86,
  },
});
