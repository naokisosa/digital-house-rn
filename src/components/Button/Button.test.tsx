import 'react-native';
import React from 'react';

import Button from './Button';
import renderer from 'react-test-renderer';

describe('Test Button component', () => {
  it('renders correctly with title', () => {
    const { toJSON } = renderer.create(<Button title="Mock" onPress={jest.fn} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with disabled', () => {
    const { toJSON } = renderer.create(<Button title="Mock" onPress={jest.fn} disabled />);
    expect(toJSON()).toMatchSnapshot();
  });
});
