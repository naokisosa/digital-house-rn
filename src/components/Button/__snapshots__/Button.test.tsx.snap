// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component renders correctly with disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Avenir",
        },
        {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 21.86,
        },
      ]
    }
  >
    Mock
  </Text>
</View>
`;

exports[`Test Button component renders correctly with title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Avenir",
        },
        {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 21.86,
        },
      ]
    }
  >
    Mock
  </Text>
</View>
`;
